imgurdownloader/imgurdownloader.py

```diff
@@ -135,29 +138,66 @@ def __init__(self, imgur_url, dir_download=os.getcwd(), file_name='',
         if self.debug:
             print('album_title: ' + self.album_title)  # debug
 
-        # get section from html that contains image ID(s) and file extensions of each ID
+        image_ids = self._init_image_ids_with_regex(html=html)
+        if image_ids:
+            self.imageIDs = image_ids
+
+        json_image_ids = list(self._init_image_ids_with_json(html=html))
+        if json_image_ids:
+            self.json_imageIDs = json_image_ids
+
+        if self.debug:
+            print("imageIDs count: %s" % str(len(self.imageIDs)))  # debug
+            print("imageIDs:\n%s" % str(self.imageIDs))  # debug
+
+        self.cnt = Counter()
+        for i in self.imageIDs:
+            self.cnt[i[1]] += 1
+
+    def _init_image_ids_with_json(self, html):
+        """get section from html that contains image ID(s) and file extensions of each ID with json."""
+        """Format of the search variable.
+        item: <java dict>\n};
+        """
+        image_ids = None
+        search_failed = False
+        search = re.search('(item:.*?};)', html, flags=re.DOTALL)
+        if search:
+            try:
+                search = search.group().replace('\n', '', ).split(':', 1)[1].rsplit('}', 1)[0]
+                json_search = json.loads(search)
+                img_dicts = json_search['album_images']['images']
```

Comment:
So, you're converting part of the html to dict via json instead of using regex to parse it? That's fine by me, but it looks like you're only considering album imgur links?

e.g.: view-source:http://imgur.com/FVRUGe2

This e.g. would fail to match any ids with your json conversion and search I believe.